### ISSUE TITLE
large performance regression in obtaining sparse array from a transpose of a sparse array

### ISSUE THREAD

**Issue by ExpandingMan** [JuliaLang/julia#28948]
```julia
using Compat.SparseArrays, Compat.Random
using BenchmarkTools

const N = 10^4

const Nnz = 5*10^6

A = zeros(Float64, N,N)

for i ∈ 1:Nnz
    A[rand(1:size(A,1)),rand(1:size(A,2))] = 1.0
end

A = sparse(A)
```
On 0.6.4:
```julia
julia> @btime sparse(transpose(A));
  292.237 ms (15 allocations: 149.00 MiB)
```

On 1.0.0
```julia
julia> @btime sparse(transpose(A));
  9.729 s (40 allocations: 240.79 MiB)
```

This scales very badly with the size of the array.  This makes life extremely difficult if you want to get the "`I, J, V`" values of the resulting transpose.

In the meantime any advice on a work around would be hugely appreciated.

**Comment by KristofferC** [comment]
Using `SparseMatrixCSC(A)`, should be fast (https://github.com/JuliaLang/julia/pull/28049). Probably need to hook up `sparse` in the same way.

**Comment by ExpandingMan** [comment]
Ah yes!  I realized that literally just 2 minutes before you posted this, thanks!

That's a huge relief, it might have been a lot of work to get around this.  Ok, I'll try to make a PR if the fix is straight-forward enough.

**Comment by KristofferC** [comment]
Fixed by #28954
